usd/src/usd_parser/USDMaterial.cc

```diff
@@ -39,18 +39,41 @@ namespace sdf
   /// \param[in] _dest The destination for the copy of _ori
   /// \return A list of UsdErrors. An empty list means no errors occurred when
   /// copying _ori to _dest
-  UsdErrors copyFile(const std::string &_ori, const std::string &_dest)
+  UsdErrors copyFile(const std::string &_ori, std::string &_dest)
   {
     UsdErrors errors;
     if (ignition::common::exists(_ori))
     {
-      // If the file exists then we should remove it and we will copy the
-      // file below
-      if (ignition::common::exists(_dest))
+      // If the file exists then we append a number suffix to the destination
+      // file
+      // For example:
+      // /bar/foo.extension
+      // /bar/foo_X.extension
+
+      std::string tmpDest = _dest;
+      std::string parentPath = ignition::common::parentPath(_dest);
+      std::string::size_type fileExtensionIndex = tmpDest.rfind(".");
+
+      std::string fileExtension = "." + tmpDest.substr(fileExtensionIndex + 1);
```

Comment:
I think that we should check the value of `fileExtensionIndex`. If the destination string being searched for does not have a `.` character, `fileExtensionIndex` would be `std::npos`, which means that the `substr` call after `rfind` would throw an exception.